### PR TITLE
Disable browserslist warning

### PR DESCRIPTION
**Pull request by timneutkens**
We saw some reports from users getting this random message that they couldn't fix.

### CODE REVIEW

**Review by leerob — APPROVED** [review]

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/browserslist-unactionable-warning | Change |
| - | - | - | - |
| buildDuration | 11.4s | 11.5s | ⚠️ +66ms |
| nodeModulesSize | 75 MB | 75 MB | ⚠️ +350 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/browserslist-unactionable-warning | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.367 | 2.313 | -0.05 |
| / avg req/sec | 1056.24 | 1080.93 | +24.69 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.632 | 1.572 | -0.06 |
| /error-in-render avg req/sec | 1531.51 | 1590.31 | +58.8 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/browserslist-unactionable-warning | Change |
| - | - | - | - |
| 677f882d2ed8..38ed.js gzip | 13.1 kB | 13.1 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-HASH.js gzip | 6.63 kB | 6.63 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.4 kB | 59.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/browserslist-unactionable-warning | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/browserslist-unactionable-warning | Change |
| - | - | - | - |
| _app-2a09aa2..4a98.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-8b758..aef6.js gzip | 3.46 kB | 3.46 kB | ✓ |
| hooks-c71ae4..70cd.js gzip | 887 B | 887 B | ✓ |
| index-bbee2f..528b.js gzip | 227 B | 227 B | ✓ |
| link-7faf09b..eba4.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..bf84.js gzip | 303 B | 303 B | ✓ |
| withRouter-a..5826.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.09 kB | 8.09 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/browserslist-unactionable-warning | Change |
| - | - | - | - |
| _buildManifest.js gzip | 321 B | 321 B | ✓ |
| Overall change | 321 B | 321 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/browserslist-unactionable-warning | Change |
| - | - | - | - |
| index.html gzip | 615 B | 615 B | ✓ |
| link.html gzip | 620 B | 620 B | ✓ |
| withRouter.html gzip | 608 B | 608 B | ✓ |
| Overall change | 1.84 kB | 1.84 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/browserslist-unactionable-warning | Change |
| - | - | - | - |
| buildDuration | 13.8s | 13.4s | -322ms |
| nodeModulesSize | 75 MB | 75 MB | ⚠️ +350 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/browserslist-unactionable-warning | Change |
| - | - | - | - |
| 677f882d2ed8..38ed.js gzip | 13.1 kB | 13.1 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-HASH.js gzip | 6.63 kB | 6.63 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.4 kB | 59.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/browserslist-unactionable-warning | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/browserslist-unactionable-warning | Change |
| - | - | - | - |
| _app-2a09aa2..4a98.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-8b758..aef6.js gzip | 3.46 kB | 3.46 kB | ✓ |
| hooks-c71ae4..70cd.js gzip | 887 B | 887 B | ✓ |
| index-bbee2f..528b.js gzip | 227 B | 227 B | ✓ |
| link-7faf09b..eba4.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..bf84.js gzip | 303 B | 303 B | ✓ |
| withRouter-a..5826.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.09 kB | 8.09 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/browserslist-unactionable-warning | Change |
| - | - | - | - |
| _buildManifest.js gzip | 321 B | 321 B | ✓ |
| Overall change | 321 B | 321 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/browserslist-unactionable-warning | Change |
| - | - | - | - |
| _error.js | 1 MB | 1 MB | ✓ |
| 404.html | 2.67 kB | 2.67 kB | ✓ |
| hooks.html | 1.92 kB | 1.92 kB | ✓ |
| index.js | 1 MB | 1 MB | ✓ |
| link.js | 1.06 MB | 1.06 MB | ✓ |
| routerDirect.js | 1.05 MB | 1.05 MB | ✓ |
| withRouter.js | 1.05 MB | 1.05 MB | ✓ |
| Overall change | 5.19 MB | 5.19 MB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Webpack 5 Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/browserslist-unactionable-warning | Change |
| - | - | - | - |
| buildDuration | 14s | 13.9s | -75ms |
| nodeModulesSize | 75 MB | 75 MB | ⚠️ +350 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/browserslist-unactionable-warning | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.378 | 2.295 | -0.08 |
| / avg req/sec | 1051.18 | 1089.17 | +37.99 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.589 | 1.591 | 0 |
| /error-in-render avg req/sec | 1573.09 | 1571.26 | ⚠️ -1.83 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/browserslist-unactionable-warning | Change |
| - | - | - | - |
| 597-7c719119..27b6.js gzip | 13 kB | 13 kB | ✓ |
| framework.HASH.js gzip | 39.3 kB | 39.3 kB | ✓ |
| main-HASH.js gzip | 6.58 kB | 6.58 kB | ✓ |
| webpack-HASH.js gzip | 954 B | 954 B | ✓ |
| Overall change | 59.9 kB | 59.9 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/browserslist-unactionable-warning | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.1 kB | 31.1 kB | ✓ |
| Overall change | 31.1 kB | 31.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/browserslist-unactionable-warning | Change |
| - | - | - | - |
| _app-6e0a903..b885.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error-7a100..a392.js gzip | 3.38 kB | 3.38 kB | ✓ |
| hooks-f7e199..11b9.js gzip | 904 B | 904 B | ✓ |
| index-3dc22c..ffbb.js gzip | 232 B | 232 B | ✓ |
| link-c010d28..b60a.js gzip | 1.63 kB | 1.63 kB | ✓ |
| routerDirect..323a.js gzip | 308 B | 308 B | ✓ |
| withRouter-a..2ef8.js gzip | 304 B | 304 B | ✓ |
| Overall change | 8.02 kB | 8.02 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/browserslist-unactionable-warning | Change |
| - | - | - | - |
| _buildManifest.js gzip | 299 B | 299 B | ✓ |
| Overall change | 299 B | 299 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | timneutkens/next.js fix/browserslist-unactionable-warning | Change |
| - | - | - | - |
| index.html gzip | 585 B | 585 B | ✓ |
| link.html gzip | 592 B | 592 B | ✓ |
| withRouter.html gzip | 579 B | 579 B | ✓ |
| Overall change | 1.76 kB | 1.76 kB | ✓ |

</details>

#### Diffs
<details>
<summary>Diff for <strong>index.html</strong></summary>

```diff
@@ -43,7 +43,7 @@
         "props": { "pageProps": {} },
         "page": "/",
         "query": {},
-        "buildId": "Z6Z_8eM1sVccvdV6IJDH9",
+        "buildId": "wnH_8cOvoVsvNclTUisUO",
         "isFallback": false,
         "gip": true
       }
@@ -77,11 +77,11 @@
       async=""
     ></script>
     <script
-      src="/_next/static/Z6Z_8eM1sVccvdV6IJDH9/_buildManifest.js"
+      src="/_next/static/wnH_8cOvoVsvNclTUisUO/_buildManifest.js"
       async=""
     ></script>
     <script
-      src="/_next/static/Z6Z_8eM1sVccvdV6IJDH9/_ssgManifest.js"
+      src="/_next/static/wnH_8cOvoVsvNclTUisUO/_ssgManifest.js"
       async=""
     ></script>
   </body>
```
</details>
<details>
<summary>Diff for <strong>link.html</strong></summary>

```diff
@@ -48,7 +48,7 @@
         "props": { "pageProps": {} },
         "page": "/link",
         "query": {},
-        "buildId": "Z6Z_8eM1sVccvdV6IJDH9",
+        "buildId": "wnH_8cOvoVsvNclTUisUO",
         "isFallback": false,
         "gip": true
       }
@@ -82,11 +82,11 @@
       async=""
     ></script>
     <script
-      src="/_next/static/Z6Z_8eM1sVccvdV6IJDH9/_buildManifest.js"
+      src="/_next/static/wnH_8cOvoVsvNclTUisUO/_buildManifest.js"
       async=""
     ></script>
     <script
-      src="/_next/static/Z6Z_8eM1sVccvdV6IJDH9/_ssgManifest.js"
+      src="/_next/static/wnH_8cOvoVsvNclTUisUO/_ssgManifest.js"
       async=""
     ></script>
   </body>
```
</details>
<details>
<summary>Diff for <strong>withRouter.html</strong></summary>

```diff
@@ -43,7 +43,7 @@
         "props": { "pageProps": {} },
         "page": "/withRouter",
         "query": {},
-        "buildId": "Z6Z_8eM1sVccvdV6IJDH9",
+        "buildId": "wnH_8cOvoVsvNclTUisUO",
         "isFallback": false,
         "gip": true
       }
@@ -77,11 +77,11 @@
       async=""
     ></script>
     <script
-      src="/_next/static/Z6Z_8eM1sVccvdV6IJDH9/_buildManifest.js"
+      src="/_next/static/wnH_8cOvoVsvNclTUisUO/_buildManifest.js"
       async=""
     ></script>
     <script
-      src="/_next/static/Z6Z_8eM1sVccvdV6IJDH9/_ssgManifest.js"
+      src="/_next/static/wnH_8cOvoVsvNclTUisUO/_ssgManifest.js"
       async=""
     ></script>
   </body>
```
</details>
</details>
Commit: b9adb4ce2f55b464aa6152d949a547c0e4918cc3

**Comment by ai** [comment]
This is an important warning. It means that Can I Use database is more than 6 months old. As the result, `last 2 version` in Babel and Autoprefixer will return old browsers. Autoprefixer and Babel will generate unnecessary polyfills, which will affect performance.

**Comment by SleeplessByte** [comment]
Except that it isn't @ai

```text
Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db
```

So let's check that out:

```bash
$ npm list caniuse-lite

`-- next@10.0.5
  +-- browserslist@4.14.6
  | `-- caniuse-lite@1.0.30001181  deduped
  +-- caniuse-lite@1.0.30001181
  `-- cssnano-simple@1.2.1
    `-- cssnano-preset-simple@1.2.1
      `-- caniuse-lite@1.0.30001181  deduped
```

`1.0.30001181` is the latest. But let's try it anyway:

```bash
$ npx browserslist@latest --update-db
npx: installed 6 in 5.173s
Current version: 1.0.30001181
New version:     1.0.30001181
Removing old caniuse-lite from lock file
Installing new caniuse-lite version

$ npm install caniuse-lite
[...]

Cleaning package.json dependencies from caniuse-lite
$ npm uninstall caniuse-lite
[...]

caniuse-lite has been successfully updated

No target browser changes
```

**Comment by ai** [comment]
@SleeplessByte can you give me a repo with error reproduction?

Maybe we have an issue with `caniuse-lite` version detection (and we should fix it and do not disable warnings).

**Comment by SleeplessByte** [comment]
I agree, I never intended warnings to be disabled. I think next.js/learn should be updated to explain what browserlist is and what happens when that command is run. I _also_ think this issue should be fixed and that they're separate issues 🗡️ 😄 

To reproduce

```bash
$ node -v
v12.20.1

$ npm -v
6.14.10

$ npx create-next-app nextjs-blog --use-npm --example "https://github.com/vercel/next-learn-starter/tree/master/learn-starter"
```

It being Windows might be related; but this happens `cmd.exe` and `git bash`.

**Comment by weyert** [comment]
@SleeplessByte I am also seeing this issue on macOS 10.15.7 with node 14.8.0.

**Comment by SleeplessByte** [comment]
Thanks. Then you just established that this is cross-platform and not Windows related. 🥇

**Comment by ai** [comment]
@SleeplessByte this warning came from `cssnano-preset-simple/dist/index.js`, which contains a huge JS file of bundled Browserslist and old Can I Use database

Since `caniuse-lite` in `cssnano-preset-simple/dist/index.js` is not part of npm, you can’t update it with `browserslist --update-db`.

**Comment by SleeplessByte** [comment]
Thanks for tracking that down. I would propose various resolutions here:

- the dependency (tree) is updated so the starter can be fixed
- perhaps browserlist can detect this and explain? 

I'm happy to PR everything myself if there is no bandwidth for these changes.

**Comment by ai** [comment]
@SleeplessByte bundling all dependencies in a single file is an anti-pattern (especially, if you do not update the dependencies).

It is better to avoid `cssnano-preset-simple` dependency or to create an issue there.

**Comment by Timer** [comment]
This has already been fixed on `next@canary`, the dependency was inlined in a few places and we made sure it got unbundled.

Closing this since it's fixed!

**Comment by SleeplessByte** [comment]
> This has already been fixed on `next@canary`, the dependency was inlined in a few places and we made sure it got unbundled.
> 
> Closing this since it's fixed!

Except that the underlying issue that started this is not fixed:

The message about browserlist is foreign to new users. I recommend updating the next.js/learn documentation to talk about this.

**Comment by SleeplessByte** [comment]
> @SleeplessByte bundling all dependencies in a single file is an anti-pattern (especially, if you do not update the dependencies).
> 
> It is better to avoid `cssnano-preset-simple` dependency or to create an issue there.

I understand that. But having a non-actionable warning seems like an issue on itself.

**Comment by ai** [comment]
> I understand that. But having a non-actionable warning seems like an issue on itself.

Of course, it should be fixed. But in a proper way. Hiding warning is fixing symptoms rather than a source of the problem.

This warning was very useful for use since it helped to find another problem.

> The message about browserlist is foreign to new users.

When `cssnano-preset-simple` bundle will be removed from dependencies, users will see this warning only after a 6 month of not updating `node_modules`.

**Comment by SleeplessByte** [comment]
> Of course, it should be fixed. But in a proper way. Hiding warning is fixing symptoms rather than a source of the problem.

I agree.

> This warning was very useful for use since it helped to find another problem.

I agree.

> It is better to avoid `cssnano-preset-simple` dependency or to create an issue there.

I agree.

Again. I never said "turn off the warning". It was a resolution by Tim, which I understand but don't agree with. I simply reported that there is a message given by `browserlist` that can't be resolved by users. It's off-putting for new users. It's not actionable. I'm glad the underlying issue has now been found: it happens when `caniuse` is bundled. As an experienced user this makes complete sense to me.

However, this issue is now closed as "Fixed in a next version". It doesn't address:
- new users running into this warning
- some other package bundling the dependency and browserlist saying it can update the definitions when it can't.

It will be fixed for the starter when `cssnano-preset-simple` is removed, unless someone downloads the starter and continues it in >= 6 months time. Or continues the project in >= 6 months time. It's just not a great experience for unexperienced people.

I don't really know how to word my concerns differently. I don't mean to be annoying or some other form of negative emotion.

**Comment by Timer** [comment]
Let's summarize really quick:

Next.js itself and one of its sub-dependencies accidentally bundled `browserslist` / `caniuse-lite`. This was **an accident** that has been resolved in both packages:

- [Next.js fix](https://github.com/vercel/next.js/pull/21515/files#diff-9d85a2d25346a67c95d98aa8d81b4d677b90708bdc40e03488ce0586edaade7eR39-R42)
- [`cssnano-preset-simple` fix](https://github.com/Timer/cssnano-preset-simple/commit/8cb758b589ee5c8095beb05e828f4fc3adb5108c)

To prevent this from happening in the future, we're going to add support for wildcard ncc externals, [tracked here](https://github.com/vercel/ncc/issues/647).

tl;dr neither `next@canary` nor `cssnano-preset-simple@latest` bundle `caniuse-lite` anymore, so the upgrade command will work fine.

Running `yarn upgrade` or `npx browserslist@latest --update-db` will fix the problem after 6 months.

---

> Is there now an issue to explain what browserlists is / does in the nextjs learn documentation?

Next.js already has documentation on [how we apply browserslist, here](https://nextjs.org/docs/advanced-features/customizing-postcss-config#customizing-target-browsers).

This material does not belong in Learn as it's not necessary for a beginner to know about. re: The fact that this message is shown on a new app has already been fixed on `next@canary`.

---

> Is there now an issue to track hardcoded/bundled caniuselite-db resolution in browserlist?

I'm not sure what issue you'd be referring to, and on what project? Can you elaborate more?

**Comment by SleeplessByte** [comment]
Thanks for that summary @Timer. 

> This material does not belong in Learn as it's not necessary for a beginner to know about. re: The fact that this message is shown on a new app has already been fixed on `next@canary`.

Great. Thanks, this is exactly what I hoped for 🥇 

> Next.js already has documentation on how we apply browserslist, here.

I understand this page because I've used it for years, I also understand it's "advanced features" so we can expect people to have experience. However, getting the notification in your terminal without any notion that that might show up, ever, is what I get questions about from people I mentor.

Don't have a good solution here, just wrote this down so you know where I'm coming from.

> I'm not sure what issue you'd be referring to, and on what project? Can you elaborate more?

The fact that `browserlists` _says_ it can update `caniuse` when it can't. Basically the first sentence of https://github.com/vercel/next.js/pull/21658#issuecomment-769913006.
